src/lib/badges/Badge.svelte

```diff
@@ -31,36 +29,91 @@
     $$props.class
   );
+
+  const closeBtnBaseClass: string = 'inline-flex items-center !p-0.5 !m-0 !ml-2 text-sm bg-transparent rounded-sm focus:!ring-0';
+
+  // uncomment the following lines to completely overwrite <CloseButton>'s color palette
+  // in case <CloseButton>'s color palette is changed in the future
+  // const closeBtnColors = {
+  //   blue: '!text-blue-400 hover:!bg-blue-200 hover:!text-blue-900 dark:hover:!bg-blue-800 dark:hover:!text-blue-300',
+  //   dark: '!text-gray-400 hover:!bg-gray-200 hover:!text-gray-900 dark:hover:!bg-gray-600 dark:hover:!text-gray-300',
+  //   red: '!text-red-400 hover:!bg-red-200 hover:!text-red-900 dark:hover:!bg-red-800 dark:hover:!text-red-300',
+  //   green: '!text-green-400 hover:!bg-green-200 hover:!text-green-900 dark:hover:!bg-green-800 dark:hover:!text-green-300',
+  //   yellow: '!text-yellow-400 hover:!bg-yellow-200 hover:!text-yellow-900 dark:hover:!bg-yellow-800 dark:hover:!text-yellow-300',
+  //   indigo: '!text-indigo-400 hover:!bg-indigo-200 hover:!text-indigo-900 dark:hover:!bg-indigo-800 dark:hover:!text-indigo-300',
+  //   purple: '!text-purple-400 hover:!bg-purple-200 hover:!text-purple-900 dark:hover:!bg-purple-800 dark:hover:!text-purple-300',
+  //   pink: '!text-pink-400 hover:!bg-pink-200 hover:!text-pink-900 dark:hover:!bg-pink-800 dark:hover:!text-pink-300'
+  // };
+
+  // only overwrite necessary colors (<CloseButton> v0.29.10)
+  const closeBtnColors = {
+    blue: 'text-blue-400 hover:text-blue-900 dark:hover:!bg-blue-800 dark:hover:text-blue-300',
+    dark: 'text-gray-400 hover:!text-gray-400 hover:!bg-gray-200 dark:hover:!text-gray-300',
+    red: 'text-red-400 hover:text-red-900 dark:hover:!bg-red-800 dark:hover:text-red-300',
+    green: 'text-green-400 hover:text-green-900 dark:hover:!bg-green-800 dark:hover:text-green-300',
+    yellow: 'text-yellow-400 hover:text-yellow-900 dark:hover:!bg-yellow-800 dark:hover:text-yellow-300',
+    indigo: 'text-indigo-400 hover:text-indigo-900 dark:hover:!bg-indigo-800 dark:hover:text-indigo-300',
+    purple: 'text-purple-400 hover:text-purple-900 dark:hover:!bg-purple-800 dark:hover:text-purple-300',
+    pink: 'text-pink-400 hover:text-pink-900 dark:hover:!bg-pink-800 dark:hover:text-pink-300'
+  };
+
+  let closeBtnClass: string;
+  $: closeBtnClass = classNames(
+    closeBtnBaseClass,
+    closeBtnColors[color]
+  );
+
+  let transition = false;
+  let hidden = false;
+  
   const handleHide = () => {
-    hidden = !hidden;
+    transition = true;
+    setTimeout(() => {
+      hidden = true;
+    }, 300);
   };
 </script>
 
-<svelte:element this={href ? 'a' : 'span'} {href} {...$$restProps} class={badgeClass} class:hidden>
+<svelte:element this={href ? 'a' : 'span'} {href} {...$$restProps} class={badgeClass} class:transition-opacity={transition} class:duration-300={transition} class:ease-out={transition} class:opacity-0={transition} class:hidden>
```

Comment:
This is a suggestion not a bug.
Instead of the lengthly:
 ```
class:transition-opacity={transition} class:duration-300={transition} class:ease-out={transition} class:opacity-0={transition}
```
we can use the advantage of using `classNames` and have just one line added to `badgeClass`:
```
transition && "transition-opacity duration-300 ease-out opacity-0",
```